sentry_streams/sentry_streams/k8s/generate.py

```diff
@@ -0,0 +1,197 @@
+        deployment["metadata"]["name"] = f"{pipeline_name}-{segid}"
+
+        volumes = deployment["spec"]["template"]["spec"].setdefault("volumes", [])
+        volumes.append(cm_volume)
+
+        for container in deployment["spec"]["template"]["spec"]["containers"]:
+            if container["name"] == container_name:
+                container["image"] = image
+
+                env = container.get("env", [])
+                env.append({"name": "SEGMENT_ID", "value": str(segid)})
+                container["env"] = env
+
+                volume_mounts = container.setdefault("volumeMounts", [])
+                volume_mounts.append(cm_volume_mount)
+
+                container["args"] = [
+                    "--config",
+                    cm_volume_mount["mountPath"],
+                ]
+
+        deployments.append(deployment)
+
+    return deployments
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Generate k8s resources from a sentry_streams deployment config."
+    )
+    parser.add_argument(
+        "--config",
+        type=argparse.FileType("r"),
+        help="Path to a deployment config file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--deployment-template",
+        type=argparse.FileType("r"),
+        help="Path to a deployment template file.",
+        default=importlib.resources.files("sentry_streams")
+        .joinpath("k8s/templates/deployment.yaml")
+        .open("r"),
+    )
+    parser.add_argument(
+        "--configmap-template",
+        type=argparse.FileType("r"),
+        help="Path to a configmap template file.",
+        default=importlib.resources.files("sentry_streams")
+        .joinpath("k8s/templates/configmap.yaml")
+        .open("r"),
+    )
+    parser.add_argument(
+        "--output",
+        type=argparse.FileType("w"),
+        help="Output target file. Defaults to stdout.",
+        default=sys.stdout,
+    )
+    parser.add_argument(
+        "--container-name",
+        type=str,
+        help="Streams segment application container name.",
+        default="segment",
+    )
+    parser.add_argument(
+        "--image",
+        type=str,
+        help="Segment container image.",
+        required=True,
+    )
+    parser.add_argument(
+        "--namespace",
+        type=str,
+        help="Namespace for deployments and configmap.",
+    )
+    args = parser.parse_args()
+
+    config = yaml.safe_load(args.config)
+    deployment_template = yaml.safe_load(args.deployment_template)
+    configmap_template = yaml.safe_load(args.configmap_template)
+
+    k8s_resources = generate_deployments(
+        config=config,
+        deployment_template=deployment_template,
+        image=args.image,
+        container_name=args.container_name,
+    ) + [
+        generate_configmap(
+            config=config,
+            configmap_template=configmap_template,
+        )
+    ]
```

Comment:
I think you may have to generate and apply the configmaps before the deployments.